MODULE.bazel

```diff
@@ -142,6 +142,12 @@ use_repo(llvm, "llvm_18_toolchain", "llvm_18_toolchain_llvm")
 
 register_toolchains("@llvm_18_toolchain//:all")
 
+# ====================================
+# Go Toolchain
+# ====================================
+go_sdk = use_extension("@rules_go//go:extensions.bzl", "go_sdk")
+go_sdk.download(version = "1.23.0")
```

Comment:
does the `go_deps` usage automatically use this?

Reply:
No that's for third party golang code. Not the golang toolchain/sdk itself. Now Bazel provides a version.

Reply:
> By default, if none of the above rules are used, the [go_register_toolchains](https://github.com/bazelbuild/rules_go/blob/master/go/toolchains.rst#go-register-toolchains) function creates a repository named @go_sdk using [go_download_sdk](https://github.com/bazelbuild/rules_go/blob/master/go/toolchains.rst#go-download-sdk), using a recent version of Go for the host operating system and architecture.

No we are specifying/pinning a version instead of letting rules_go pick.